homeassistant/components/ezviz/select.py

```diff
@@ -0,0 +1,98 @@
+"""Support for EZVIZ select controls."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pyezviz.constants import DeviceSwitchType, SoundMode
+from pyezviz.exceptions import HTTPError, PyEzvizError
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DATA_COORDINATOR, DOMAIN
+from .coordinator import EzvizDataUpdateCoordinator
+from .entity import EzvizEntity
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass
+class EzvizSelectEntityDescriptionMixin:
+    """Mixin values for EZVIZ Select entities."""
+
+    supported_switch: int
+
+
+@dataclass
+class EzvizSelectEntityDescription(
+    SelectEntityDescription, EzvizSelectEntityDescriptionMixin
+):
+    """Describe a EZVIZ Select entity."""
+
+
+SELECT_TYPE = EzvizSelectEntityDescription(
+    key="alarm_sound_mod",
+    translation_key="alarm_sound_mode",
+    icon="mdi:alarm",
+    entity_category=EntityCategory.CONFIG,
+    options=["soft", "intensive", "silent"],
+    supported_switch=DeviceSwitchType.ALARM_TONE.value,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up EZVIZ select entities based on a config entry."""
+    coordinator: EzvizDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    async_add_entities(
+        EzvizSensor(coordinator, camera)
+        for camera in coordinator.data
+        for switch in coordinator.data[camera]["switches"]
+        if switch == SELECT_TYPE.supported_switch
+    )
+
+
+class EzvizSensor(EzvizEntity, SelectEntity):
+    """Representation of a EZVIZ select entity."""
+
+    _attr_has_entity_name = True
```

Comment:
Can probably set this on `EzvizEntity` and remove it from the entities inheriting it.

Reply:
Have pull requests open for nearly every entity. Plan was to open another pull request after they are all done to review EzvizEntity and remove any duplicates.